src/Features/CSharp/Portable/Completion/KeywordRecommenders/WhenKeywordRecommender.cs

```diff
@@ -14,7 +14,31 @@ public WhenKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            return context.IsCatchFilterContext;
+            if (context.IsCatchFilterContext)
+            {
+                return true;
+            }
+
+            // case int i $$
+            if (IsIdentifierInCasePattern(context.LeftToken.Parent))
+            {
+                return true;
+            }
+
+            // case int i w$$
+            if (IsIdentifierInCasePattern(context.LeftToken.GetPreviousToken().Parent))
```

Comment:
this shouldn't be needed.  see other recommenders for how they handle both the unwritten and partially written case.

Reply:
I think you can jsut use "context.TargetToken" for this.